src/supernova/test.rs

```diff
@@ -178,14 +179,15 @@ where
   fn synthesize<CS: ConstraintSystem<F>>(
     &self,
     cs: &mut CS,
-    pc: &AllocatedNum<F>,
+    pc: Option<&AllocatedNum<F>>,
     z: &[AllocatedNum<F>],
-  ) -> Result<(AllocatedNum<F>, Vec<AllocatedNum<F>>), SynthesisError> {
+  ) -> Result<(Option<AllocatedNum<F>>, Vec<AllocatedNum<F>>), SynthesisError> {
     // constrain rom[pc] equal to `self.circuit_index`
     let circuit_index = alloc_const(
       cs.namespace(|| "circuit_index"),
       F::from(self.circuit_index as u64),
     )?;
+    let pc = pc.unwrap();
```

Comment:
An expect wouldn’t go amiss, as it could explain that you intend this `SquareCircuit` to always be used on the primary.